### PR TITLE
[WIP] core.py: encode utf-8 when streaming logs

### PR DESCRIPTION
**Pull request by vrutkovs**
Signed-off-by: Vadim Rutkovsky <vrutkovs@redhat.com>

### CODE REVIEW

File: osbs/core.py [hunk 1]
```diff
@@ -494,7 +494,12 @@ def logs(self, build_id, follow=False, build_json=None, wait_if_missing=False):
         buildlogs_url = self._build_url("builds/%s/log/" % build_id)
         response = self._get(buildlogs_url, headers={'Connection': 'close'})
         check_response(response)
-        return response.content
+
+        content = response.content
+        if isinstance(content, bytes):
+            return content
+        else:
+            return content.encode('utf-8')
```
Comment (twaugh) [inline osbs/core.py]: Why do we need to encode it this early? We shouldn't encode it until we need to serialize it.

**Comment by lcarva** [comment]
I think reverting https://github.com/projectatomic/osbs-client/commit/8182bfdef54f0968c950c27503afe6227b5bfa33 should do the trick.

**Comment by twaugh** [comment]
Yes. iter_lines() returns a byte sequence:
http://docs.python-requests.org/en/master/user/advanced/?highlight=iter_lines#streaming-requests

so we need to discover its encoding (it's JSON so we can do that automatically) and decode it.

**Comment by vrutkovs** [comment]
It doesn't work on Py3 and there are alternative solutions. Closing

**Pull request closed without merging**